### ISSUE TITLE
Docker images broken on Dockerhub

### ISSUE THREAD

**Issue by rbeucher** [underworldcode/underworld2#225]
The docker images on docker hub are broken. Build started to fail one month ago...

**Comment by jmansour** [comment]
Ok, the issue at the underworld2_tested stage is resolved, and these images are now successfully being created:
https://hub.docker.com/r/underworldcode/underworld2_untested/builds/

However there's now an issue relating to visualisation that's preventing the tests passing and therefore the dev tested images are still failing:
https://hub.docker.com/r/underworldcode/underworld2/builds/

@OKaluza, can you have a look at this when you get a chance?  I've had a quick play, and seemingly there's an issue relating to adding labels/etc to images.  You can get to the point of failure by running
```bash
docker run -it --rm underworldcode/underworld2_untested:dev /bin/bash
```
and then from within the container
```bash
cd /root/underworld2/docs/user_guide
runipy 08_Visualisation.ipynb
```

which bombs out for me with 
```
fig.show()

SQL prepare error: (insert into geometry (object_id, timestep, rank, idx, type, data_type, size, count, width, minimum, maximum, dim_factor, units, minX, minY, minZ, maxX, maxY, maxZ, labels, data) values (1, 0, 0, 0, 0, 0, 3, 3, 0, 0, 1, 0, '', 1.4013e-45, 0, -nan, -nan, 0, 2.98443e+19, ?, ?)) no such column: nan

05/31/2017 05:43:42 AM INFO: Cell returned
05/31/2017 05:43:42 AM INFO: Running cell:
fig.save_image("highdef.png", (1600,0))

SQLite Issue: database schema is locked: main
delete from geometry where timestep between 0 and 0; delete from timestep where id between 0 and 0;
SQLite Issue: database schema is locked: main
insert into timestep (id, time, properties) values (0, 0, '')
SQLite Issue: database schema is locked: main
create table IF NOT EXISTS state (id INTEGER PRIMARY KEY ASC, name VARCHAR(256), data TEXT)
SQLite Issue: database schema is locked: main
delete from state where name == 'Figure_8'
SQL prepare error: (insert into state (name, data) values ('Figure_8', ?)) database schema is locked: main
SQL prepare error: (insert into geometry (object_id, timestep, rank, idx, type, data_type, size, count, width, minimum, maximum, dim_factor, units, minX, minY, minZ, maxX, maxY, maxZ, labels, data) values (1, 0, 0, 0, 0, 0, 3, 3, 0, 0, 1, 0, '', 1.4013e-45, 0, -nan, -nan, 0, 2.98443e+19, ?, ?)) no such column: nan

05/31/2017 05:43:42 AM INFO: Cell raised uncaught exception:
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
<ipython-input-22-8f2ff4a6ee06> in <module>()
----> 1 fig.save_image("highdef.png", (1600,0))

/root/underworld2/glucifer/_glucifer.pyc in save_image(self, filename, size)
    577     def save_image(self, filename="", size=(0,0)):
    578         # For back compatibility
--> 579         return self.save(filename, size)
    580
    581     def save_database(self, filename, regen=True):

/root/underworld2/glucifer/_glucifer.pyc in save(self, filename, size, type)
    604             anything.
    605         """
--> 606         self._generate_DB()
    607         global lavavu
    608         if filename is None or not lavavu or uw.rank() > 0:

/root/underworld2/glucifer/_glucifer.pyc in _generate_DB(self)
    630         if len(self.draw.vertices) > 0:
    631             objects.append(self.draw)
--> 632         self.db._generate(self.name, objects, self)
    633
    634     def _generate_image(self, filename="", size=(0,0)):

/root/underworld2/glucifer/_glucifer.pyc in _generate(self, figname, objects, props)
    234                 for obj in self._objects:
    235                     #Create/Transform geometry by object
--> 236                     obj.render(lv)
    237         else:
    238             #Open db, get state and update it to match active figure

/root/underworld2/glucifer/objects.pyc in render(self, viewer)
    254         #Update the database
    255         if output:
--> 256             viewer.app.update(obj.ref, True)
    257
    258     def resetDrawing(self):

RuntimeError: SQL prepare error: (insert into geometry (object_id, timestep, rank, idx, type, data_type, size, count, width, minimum, maximum, dim_factor, units, minX, minY, minZ, maxX, maxY, maxZ, labels, data) values (1, 0, 0, 0, 0, 0, 3, 3, 0, 0, 1, 0, '', 1.4013e-45, 0, -nan, -nan, 0, 2.98443e+19, ?, ?)) no such column: nan
```

**Comment by OKaluza** [comment]
Found the problem, only occurs (sometimes) when compiled with optimisation and no debug which is why it's only showing up here.
Have to sort a few things out before I update the viewer module with the fix but should be in later today.

**Comment by OKaluza** [comment]
This should be fixed, as of 97fae7e5d27e5380dab3d01c53625ca654866b04 (important bit here https://github.com/OKaluza/LavaVu/commit/7e527e0e94dd69bf4cb6d727616c17623a1eeeb2)

**Comment by jmansour** [comment]
https://hub.docker.com/r/underworldcode/underworld2/builds/

great success!!
